### PR TITLE
perf: generate config files according to whether config content exist

### PR DESCRIPTION
**Pull request by wenytang-ms**

### CODE REVIEW

**Comment by wenytang-ms** [comment]
this PR modify 3 test cases.
1. create a tab app without sa, so there will be no any config or teamsfx in templates.
2. create a tab app without sa, then add bot plugin, there will be add config and teamsfx.
3. create a tab app with sa, there must be config and teamsfx in templates, then add bot plugin, there the config added bot plugin, and teamsfx folder add bot config bicep.

File: packages/fx-core/tests/plugins/solution/solution.arm.test.ts [hunk 1]
```diff
@@ -292,6 +239,10 @@ Mocked aad configuration orchestration content. Module path: './teamsFx/aadConfi
         AadParameter: `${TestFileContent.aadParameterValue}`,
       })
     );
+    expect(await fs.pathExists(path.join(projectArmTemplateFolder, TestFilePath.configFileName))).to
```
Comment (blackchoey) [inline packages/fx-core/tests/plugins/solution/solution.arm.test.ts:242]: should we expect this to be true? The bot plugin requires config

Comment (wenytang-ms) [reply packages/fx-core/tests/plugins/solution/solution.arm.test.ts:242]: 261 line will add bot plugin, current there has no any other plugins except frontend hosting. 
so there is no any config files

File: packages/fx-core/tests/plugins/solution/solution.arm.test.ts [hunk 2]
```diff
@@ -390,15 +334,155 @@ Mocked bot provision orchestration content. Module path: './provision/botProvisi
...
+    TestHelper.mockedFehostGenerateArmTemplates(mocker);
+    TestHelper.mockedAadGenerateArmTemplates(mocker);
+    TestHelper.mockedIdentityGenerateArmTemplates(mocker);
+    TestHelper.mockedSimpleAuthGenerateArmTemplates(mocker);
+    TestHelper.mockedBotGenerateArmTemplates(mocker);
+    TestHelper.mockedFeHostUpdateArmTemplates(mocker);
+    TestHelper.mockedIdentityUpdateArmTemplates(mocker);
+    TestHelper.mockedSimpleAuthUpdateArmTemplates(mocker);
+    // Action
+    let result = await generateArmTemplate(mockedCtx, [
```
Comment (blackchoey) [inline packages/fx-core/tests/plugins/solution/solution.arm.test.ts:366]: the second parameter is selected plugin, should it only conatins bot plugin in this test case?

Comment (blackchoey) [reply packages/fx-core/tests/plugins/solution/solution.arm.test.ts:366]: for this test case, my understanding is we're testing the situation that when config.bicep and teamsfx folder already exists, adding another resource plugin that requires config will success with no error. So maybe we can generateArmTemplates(ctx,[tab,simpleauth]) first, then generateArmTemplates(ctx, [bot]) to test this scenario.

File: packages/fx-core/tests/plugins/solution/solution.arm.test.ts (hunk 1 above)
Comment (blackchoey) [reply packages/fx-core/tests/plugins/solution/solution.arm.test.ts:242]: thanks

File: packages/fx-core/tests/plugins/solution/solution.arm.test.ts [hunk 3]
```diff
@@ -390,15 +334,155 @@ Mocked bot provision orchestration content. Module path: './provision/botProvisi
...
+      aadPlugin,
+      botPlugin,
```
Comment (blackchoey) [inline packages/fx-core/tests/plugins/solution/solution.arm.test.ts:368]: the bot plugin should be fehostPlugin here?

Comment (wenytang-ms) [reply packages/fx-core/tests/plugins/solution/solution.arm.test.ts:368]: thanks!

**Review by blackchoey — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**